controllers/reconcilers/configuration/prometheus.go

```diff
@@ -473,6 +474,38 @@ func (r *Reconciler) reconcilePrometheus(ctx context.Context, cr *v1.Observabili
 	return nil
 }
 
+//resource.Quantity should be greater than than the default 250Gi
+func validateOverridePrometheusStorageSize(q resource.Quantity, cr *v1.Observability) bool {
+	return q.CmpInt64(250) >= 0
+}
+
+//construct Promnetheus storage spec wiht either default or override value from resources
+func getPrometheusStorageSpecHelper(cr *v1.Observability, indexes []v1.RepositoryIndex) (*prometheusv1.StorageSpec, error) {
+	prometheusStorageSpec := cr.Spec.Storage.PrometheusStorageSpec
+	customStorageSize := model.GetPrometheusStorageSize(cr, indexes)
```

Comment:
with the current implementation, this can return an empty string (if no override exists).